src/messages.js

```diff
@@ -65,9 +72,111 @@
   JSAV.ext.clearumsg = function(msg, options) {
     this._msg.clear();
   };
-  
+
+  ////////////////////////
+  ///////TextToSpeech/////
+  JSAV.ext.textToSpeech = function(speechText) {
+    var modifiedMsg;
+    modifiedMsg = speechText.replace(/<[^>]*>/g, "");
+    modifiedMsg = modifiedMsg.replace(/\$/g, "");
+    modifiedMsg = modifiedMsg.replace(/\\mathbf/gi, "");
+    modifiedMsg = modifiedMsg.replace(/\\displaystyle/gi, "");
+    modifiedMsg = modifiedMsg.replace(/\\mbox/gi, "");
+    modifiedMsg = modifiedMsg.replace(/n-/gi, "n minus");
+    modifiedMsg = modifiedMsg.replace(/m-/gi, "m minus");
+    modifiedMsg = modifiedMsg.replace(/%/gi, "remainder");
+    var synth = window.speechSynthesis;
+    var u = new SpeechSynthesisUtterance();
+    var amISpeaking = synth.speaking;
+    u.lang = "en-US";
+    u.rate = 1.0;
+    // Assign the umsg text to the narration file
+    u.text = modifiedMsg;
+    if(amISpeaking === true) {
+      synth.cancel();
+    }
+    synth.speak(u);
+  };
+  //////////////////////////////////////////
+////////////Sound control button
+  JSAV.ext.soundSettings = function(jsav) {
+    var self = this;
+    // creating the button element
+    var $elem = $("<button class='jsavsound soundOff'></button>");
+    // Sound button click event
+    $elem.click(function() {
+      var txt = $(this.offsetParent).find(".jsavoutput").clone().find(".MathJax").remove().end().text();
+      JSAV.ext.textToSpeech(txt);
+      // SwitchingOff sound buttons in other frames
+      for (var j = 0; j < window.length; j++) {
+        $(".sound", window.frames[j].document).not(this).each(function() {
+          $(this.offsetParent).parent().find(".jsavcontainer").attr("voice", "false");
+          $(this).removeClass("sound").addClass("soundOff");
+        });
+      }
+      // SwitchingOff sound buttons not in frames
+      $(".sound", window.document).not(this).each(function() {
+        $(this.offsetParent).parent().find(".jsavcontainer").attr("voice", "false");
+        $(this).removeClass("sounds").addClass("soundOff");
+      });
+      //Toggle voice attribute and stop the current voice
+      var synth = window.speechSynthesis;
```

Comment:
Here you could check if the API is supported by making sure synth is defined before you start using it. Or perhaps already in the beginning of the function.